src/main/scala/nl/vroste/zio/kinesis/client/DynamicConsumer.scala

```diff
@@ -19,6 +19,7 @@ import zio.blocking.Blocking
 import zio.stream.ZStream
 
 import scala.jdk.CollectionConverters._
+import software.amazon.kinesis.exceptions.ShutdownException
 
 /**
  * Offers a ZStream based interface to the Amazon Kinesis Client Library (KCL)
```

Comment:
A top-down shutdown of the stream can still be useful for some streams. For example if you use `aggregateWithinAsync`, it will not emit an element when the fiber is interrupted. It will when you stop the source of the stream and wait for the stream to finish. 

Similar behavior is necessary in alpakka-kafka for example, https://doc.akka.io/docs/alpakka-kafka/current/consumer.html